src/Product/ProductList.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace Automattic\WooCommerce\GoogleListingsAndAds\Product;
+
+use WC_Product;
+
+/**
+ * Class ProductList
+ *
+ * A list of products returned from wc_get_products.
+ *
+ * @since x.x.x
+ *
+ * @package Automattic\WooCommerce\GoogleListingsAndAds\Product
+ */
+class ProductList {
```

Comment:
I considered that but didn't want to add too many functions to the class which we don't even use. It will be less relevant if we add a `ProductFilter` class since there will be less functions handling a filtered product list.

Reply:
I played around with trying to implement the extra interfaces. But we use strict type checking, so even though we can access the results like an array we'll still need a whole other set of changes so it allows passing the list as an object, that makes the extra `->get` calls seem less intrusive.